WeakAuras/Prototypes.lua

```diff
@@ -6931,97 +7108,121 @@ end
 
 WeakAuras.dynamic_texts = {
   ["p"] = {
-    func = function(state, region)
-      if not state then return "" end
+    get = function(state)
+      if not state then return nil end
       if state.progressType == "static" then
-        return state.value or ""
+        return state.value or nil
       end
       if state.progressType == "timed" then
         if not state.expirationTime or not state.duration then
-          return ""
+          return nil
         end
         local remaining  = state.expirationTime - GetTime();
-        local duration  = state.duration;
-
-        local remainingStr     = "";
-        if remaining == math.huge then
-          remainingStr     = " ";
-        elseif remaining > 60 then
-          remainingStr     = string.format("%i:", math.floor(remaining / 60));
-          remaining        = remaining % 60;
-          remainingStr     = remainingStr..string.format("%02i", remaining);
-        elseif remaining > 0 then
-          local progressPrecision = region.progressPrecision and math.abs(region.progressPrecision) or 1
-          -- remainingStr = remainingStr..string.format("%."..(data.progressPrecision or 1).."f", remaining);
-          if progressPrecision == 4 and remaining <= 3 then
-            remainingStr = remainingStr..string.format("%.1f", remaining);
-          elseif progressPrecision == 5 and remaining <= 3 then
-            remainingStr = remainingStr..string.format("%.2f", remaining);
-          elseif (progressPrecision == 4 or progressPrecision == 5) and remaining > 3 then
-            remainingStr = remainingStr..string.format("%d", remaining);
-          else
-            remainingStr = remainingStr..string.format("%.".. progressPrecision .."f", remaining);
-          end
+        return remaining
+      end
+    end,
+    func = function(remaining, state, progressPrecision)
+      progressPrecision = progressPrecision or 1
+      if not state or state.progressType ~= "timed" then
+        return remaining
+      end
+      if type(remaining) ~= "number" then
+        return ""
+      end
+
+      local remainingStr     = "";
+      if remaining == math.huge then
+        remainingStr     = " ";
+      elseif remaining > 60 then
+        remainingStr     = string.format("%i:", math.floor(remaining / 60));
+        remaining        = remaining % 60;
+        remainingStr     = remainingStr..string.format("%02i", remaining);
```

Comment:
i would have written it like that (negligible i guess)
`remainingStr     = ("%s%02i"):format(remainingStr, remaining);`

Reply:
I copied this code from the exstng formater.